### ISSUE TITLE
Feature: line markers with long text labels

### ISSUE THREAD

**Issue by fviolette** [expressive-code/expressive-code#128]
Hello and thanks for the "labels to line markers" feature which is very useful: https://expressive-code.com/key-features/text-markers/#adding-labels-to-line-markers

I have an example where my annotations are in the form of:

// 1. Register the ContentsquareWebViewController by mapping InAppWebView's controller
{lines of code}
// 2. Register the javascript channels
{lines of code}
etc.

If I try to port these with the current syntax...

````
```dart {"1. Register the ContentsquareWebViewController by mapping InAppWebView's controller": 8-18})
````

..the text will overlap the code.

![image](https://github.com/expressive-code/expressive-code/assets/19730806/e4e9d218-4198-4277-847f-319412c631e4)

I'm used to working in AsciiDoc, which can handle these cases as follows (see example 54 right below the [anchor target](https://docs.asciidoctor.org/asciidoc/latest/syntax-quick-reference/#literals-and-source-code)).

Is there something I missed or was this not the intent or the original version of the feature?
Do you have any idea how I could work around this without working around the feature?

Thanks!

**Comment by hippotastic** [comment]
Hehe, interesting use case!

The original intent of these labels is for them to be very short, as shown in the example in the docs:

![image](https://github.com/expressive-code/expressive-code/assets/6137925/307a387b-2789-475c-b80a-51a7be15d2b8)

So, in your case you would just make the label "1" without any additional text, and then put a numbered markdown list below your code block in markdown where you explain what to do.

It's interesting to see the AsciiDoc example which seems similar to footnotes. This could be another new feature. 😄

**Comment by fviolette** [comment]
No personal preference on foonotes vs. the inline labels.
With footnote style annotations you have to go back and forth but it might be cleaner with dense blocks.

Also, I actually skipped the note in your screenshot but I assume adjusting the setting could mess up some occurrences while fixing other ones (long vs short labels) ?

**Comment by hippotastic** [comment]
Yes, adjusting the `codePaddingInline` setting increases the space available for annotations on the sides of all code blocks. This is meant to allow slightly longer labels, e.g. 2-3 characters instead of just 1.

This would not work for your example though, because making the side paddings so large to fit these long labels would lead to a ton of wasted space in all code blocks.

I think your use case would be better solved with custom annotations that have their own line. Basically a way to convert some special comment syntax right inside a code line to a styled box. This would make a great optional plugin!

**Comment by fviolette** [comment]
Alright! Thanks for your feedback 👍 
I updated the issue description, let me know if we need to do anything for now.

**Comment by hippotastic** [comment]
Actually, I just realized while looking at the code that this is already possible with the current implementation! :) All you need to do is add an empty line at the beginning of the line range to reserve space where the label should appear. I've added an example to the docs on how to do it:

https://expressive-code.com/key-features/text-markers/#adding-long-labels-on-their-own-lines

Does this solve your use case?

**Comment by fviolette** [comment]
I found this by testing myself — found the EC docs only today and again missed that :|
If this is an official way of doing this then yes it works for me :)

Thanks again!

**Comment by hippotastic** [comment]
Don't worry about missing that, I just added this example right now based on your use case! You couldn't have found it before. 😄

I realize that adding empty lines to reserve space isn't the most elegant solution, but it works for now and will continue to work. I have a feature planned that unlocks various cool inline comment syntax possibilities that will become a better solution for this use case as well and that doesn't require you to count line numbers. Stay tuned for future versions! :)

Closing this for now as there's a solution available.
